lte/gateway/c/core/oai/test/s1ap_task/test_s1ap_mme_handlers.cpp

```diff
@@ -1991,6 +2062,185 @@ TEST_F(S1apMmeHandlersTest, HandleS1apErabModificationCnf) {
+                         0x00, 0x32, 0x40, 0x0a, 0x0a, 0x1f, 0xc0, 0xa8,
+                         0x3c, 0x8d, 0x0a, 0x00, 0x01, 0x28};
+
+  ASSERT_EQ(
+      simulate_pdu_s1_message(
+          ics_bytes, sizeof(ics_bytes), state, assoc_id, stream_id),
+      RETURNok);
+
+  // State validation
+  ASSERT_TRUE(is_ue_state_valid(assoc_id, 1, S1AP_UE_CONNECTED));
+  ASSERT_TRUE(is_enb_state_valid(state, assoc_id, S1AP_READY, 1));
+
+  // Simulate Attach Complete
+  uint8_t attach_compl_bytes[] = {
+      0x00, 0x0d, 0x40, 0x37, 0x00, 0x00, 0x05, 0x00, 0x00, 0x00, 0x02, 0x00,
+      0x07, 0x00, 0x08, 0x00, 0x02, 0x00, 0x01, 0x00, 0x1a, 0x00, 0x0e, 0x0d,
+      0x27, 0xeb, 0x9e, 0x7f, 0x7e, 0x01, 0x07, 0x43, 0x00, 0x03, 0x52, 0x00,
+      0xc2, 0x00, 0x64, 0x40, 0x08, 0x00, 0x00, 0xf1, 0x10, 0x00, 0x00, 0x00,
+      0x10, 0x00, 0x43, 0x40, 0x06, 0x00, 0x00, 0xf1, 0x10, 0x00, 0x01};
+
+  ASSERT_EQ(
+      simulate_pdu_s1_message(
+          attach_compl_bytes, sizeof(attach_compl_bytes), state, assoc_id,
+          stream_id),
+      RETURNok);
+
+  // Simulate ENBConfigurationTransfer PDU payload
+  uint8_t enb_transfer_bytes[] = {
+      0x00, 0x28, 0x40, 0x30, 0x00, 0x00, 0x01, 0x00, 0x81, 0x00, 0x29,
+      0x40, 0x00, 0xf1, 0x10, 0x40, 0x00, 0x00, 0x00, 0x20, 0x00, 0xf1,
+      0x10, 0x00, 0x02, 0x00, 0x00, 0xf1, 0x10, 0x40, 0x00, 0x00, 0x00,
+      0x10, 0x00, 0xf1, 0x10, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00, 0x98,
+      0x40, 0x06, 0x01, 0xf0, 0xc0, 0xa8, 0x3c, 0x8d};
+
+  ASSERT_EQ(
+      simulate_pdu_s1_message(
+          enb_transfer_bytes, sizeof(enb_transfer_bytes), state, assoc_id,
+          stream_id),
+      RETURNok);
+
+  // Simulate second ENBConfigurationTransfer PDU payload
+  uint8_t enb_transfer_bytes_2[] = {
+      0x00, 0x28, 0x40, 0x2a, 0x00, 0x00, 0x01, 0x00, 0x81, 0x00, 0x23, 0x00,
+      0x00, 0xf1, 0x10, 0x40, 0x00, 0x00, 0x00, 0x20, 0x00, 0xf1, 0x10, 0x00,
+      0x01, 0x00, 0x00, 0xf1, 0x10, 0x40, 0x00, 0x00, 0x00, 0x10, 0x00, 0xf1,
+      0x10, 0x00, 0x02, 0x50, 0x0f, 0x80, 0xc0, 0xa8, 0x3c, 0x8d};
+
+  ASSERT_EQ(
+      simulate_pdu_s1_message(
+          enb_transfer_bytes_2, sizeof(enb_transfer_bytes_2), state, assoc_id,
+          stream_id),
+      RETURNok);
+
+  // Simulate PathSwitchRequest PDU payload
+  uint8_t path_switch_bytes[] = {
+      0x00, 0x03, 0x00, 0x41, 0x00, 0x00, 0x06, 0x00, 0x08, 0x00, 0x02, 0x00,
+      0x02, 0x00, 0x16, 0x00, 0x0f, 0x00, 0x00, 0x17, 0x00, 0x0a, 0x0a, 0x1f,
+      0xc0, 0xa8, 0x3c, 0x8d, 0x0a, 0x00, 0x01, 0x29, 0x00, 0x58, 0x00, 0x02,
+      0x00, 0x07, 0x00, 0x64, 0x40, 0x08, 0x00, 0x00, 0xf1, 0x10, 0x00, 0x00,
+      0x00, 0x20, 0x00, 0x43, 0x40, 0x06, 0x00, 0x00, 0xf1, 0x10, 0x00, 0x02,
+      0x00, 0x6b, 0x40, 0x05, 0x18, 0x00, 0x0c, 0x00, 0x00};
+
+  ASSERT_EQ(
+      simulate_pdu_s1_message(
+          path_switch_bytes, sizeof(path_switch_bytes), state, assoc_id,
+          stream_id),
+      RETURNok);
+
```

Comment:
Can we check that UE context on the S1AP task has actually switched the eNB association id after the Path Switch Request?